### ISSUE TITLE
Error when opening a tv show folder using the website api

### ISSUE THREAD

**Issue by Paco8** [Sandmann79/xbmc#734]
**Addon used**
- [x] Amazon VOD (plugin.video.amazon-test)
- [ ] Browser Launcher (plugin.program.browser.launcher)

- Addon version: 1.0.6~beta1

**Account type**
- [x] primevideo.com
- [ ] amazon.(com/co.uk/de/jp)

**System Setup (please provide the following information):**
- Hardware: PC
- OS version: Ubuntu 22.04
- Kodi version number: 19.4

**Upload Logs**

```
2024-04-18 15:21:49.301 T:8982    DEBUG <general>: [Amazon VOD] Requested /pv/browse/root/Watchlist/watchlist/tv
2024-04-18 15:21:49.394 T:8916    DEBUG <general>: ------ Window Init (DialogBusy.xml) ------
2024-04-18 15:21:49.687 T:8982    DEBUG <general>: [Amazon VOD] Download Time: 0.3838053370000125
2024-04-18 15:21:50.181 T:8982    DEBUG <general>: [Amazon VOD] Download Time: 0.4444083739999769
2024-04-18 15:21:50.242 T:8982    ERROR <general>: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                                    - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                                   Error Type: <class 'KeyError'>
                                                   Error Contents: 'titleIds'
                                                   Traceback (most recent call last):
                                                     File "/home/paco/.kodi/addons/plugin.video.amazon-test/default.py", line 9, in <module>
                                                       EntryPoint(argv)
                                                     File "/home/paco/.kodi/addons/plugin.video.amazon-test/resources/lib/startup.py", line 67, in EntryPoint
                                                       _g.pv.Route(verb, path)
                                                     File "/home/paco/.kodi/addons/plugin.video.amazon-test/resources/lib/web_api.py", line 184, in Route
                                                       elif 'browse' == verb: self._g.pv.Browse(path)
                                                     File "/home/paco/.kodi/addons/plugin.video.amazon-test/resources/lib/web_api.py", line 458, in Browse
                                                       node, breadcrumb = self._TraverseCatalog(path)
                                                     File "/home/paco/.kodi/addons/plugin.video.amazon-test/resources/lib/web_api.py", line 154, in _TraverseCatalog
                                                       self._LazyLoad(node[nodeName], pathList[0:1 + i])
                                                     File "/home/paco/.kodi/addons/plugin.video.amazon-test/resources/lib/web_api.py", line 1383, in _LazyLoad
                                                       bUpdatedVideoData |= AddSeason(breadcrumb[-1], o, bCacheRefresh, iu)
                                                     File "/home/paco/.kodi/addons/plugin.video.amazon-test/resources/lib/web_api.py", line 839, in AddSeason
                                                       bUpdatedVideoData |= ParseSinglePage(oid, season, bCacheRefresh, url=url)
                                                     File "/home/paco/.kodi/addons/plugin.video.amazon-test/resources/lib/web_api.py", line 1016, in ParseSinglePage
                                                       for e in le['titleIds']:
                                                   KeyError: 'titleIds'
                                                   -->End of Python script error report<--
   
```

**Describe the bug**

There's that error when I try to list a tv show folder, using the website api. I guess they changed something in the website.

**Comment by PsyEng** [comment]
Change the data source to android, this should fix it. Maybe you've to login again or restart kodi/your device.

**Comment by stephematician** [comment]
Same error here.

**Comment by Paco8** [comment]
> Change the data source to android, this should fix it. Maybe you've to login again or restart kodi/your device.

Yes, it works with the android api, but it seems it's not possible to change the profile with this api.

**Comment by stefano-salari** [comment]
Same error here with LibreElec 10.0.4 (on RPi2.arm) with plugin version 1.0.5+matrix.1

**Comment by Sandmann79** [comment]
This issue has been fixed/finished with 60b881c39379a6e7a090ba582d3563b5c3ffc75d.

Just a short note regarding the new version:
Following recent changes to the website, Amazon no longer provides detailed information on all available seasons, only on the one currently selected.
This information is now only updated when a season is called up or when a show/season is refreshed manually.

**Comment by stefano-salari** [comment]
Thank you so much, I just installed the fix version and it works fine!
I'd say that my side the issue is fixed.
